src/core/config/loaders/inline_loader.cpp

```
@@ -0,0 +1,122 @@
+}
+
+std::map<std::string, std::experimental::any> inline_loader::load_all() & {
+  if (m_data.empty()) {
+    parse_inline_data();
+  }
+  return m_data;
+}
+
+void inline_loader::parse_inline_data() {
+  std::vector<std::string> pairs = split(m_inline_config, ',');
+  for (const std::string &kv : pairs) {
+    const std::string trimmed_kv = remove_spaces(kv);
+
+    if (trimmed_kv.empty()) {
+      continue;
+    }
+
+    const std::string::size_type eq_pos = trimmed_kv.find('=');
+    if (eq_pos == std::string::npos) {
+      throw_xlio_exception(errors::INLINE_LOADER_INVALID_PAIR + trimmed_kv);
+    }
+
+    std::string key = trimmed_kv.substr(0, eq_pos);
+    std::string val = trimmed_kv.substr(eq_pos + 1);
+
+    // If either the key is empty, it's invalid
+    if (trimmed_kv.empty()) {
+      throw_xlio_exception(errors::INLINE_LOADER_EMPTY_KEY + kv);
+    }
+
+    // Attempt to parse the value as bool/int, otherwise store string
+    m_data[key] = parse_value(val);
+  }
+
+  if (m_data.empty()) {
+    throw_xlio_exception(errors::INLINE_LOADER_INVALID_CONFIG +
+                         std::string(m_inline_config));
+  }
+}
+
+static std::vector<std::string> split(const std::string &s, char delimiter) {
+  std::vector<std::string> tokens;
+  std::string token;
+  std::istringstream token_stream(s);
+  while (std::getline(token_stream, token, delimiter)) {
+    tokens.push_back(token);
+  }
+  return tokens;
+}
+
+static std::string remove_spaces(std::string str) {
+  str.erase(std::remove_if(str.begin(), str.end(), ::isspace), str.end());
+  return str;
+}
+
+static std::experimental::any parse_value(const std::string &val) {
+  // Check for bool
+  if (compare_case_insensitive(val, error_format::TRUE_VALUE)) {
+    return std::experimental::any(true);
+  }
+  if (compare_case_insensitive(val, error_format::FALSE_VALUE)) {
+    return std::experimental::any(false);
+  }
+
+  try {
+    const int64_t int_val = std::stoll(val);
```

I dont like the structure of the function, maybe best to have one return at the end, also the empty catch, there are two types of exception, are both valid here?